### ISSUE TITLE
Implement proper config initialization for default values

### ISSUE THREAD

**Issue by theseion** [coreruleset/go-ftw#108]
For values that need a default other than the zero value (e.g., `20`), the configuration currently isn't initialized properly in tests, as the defaults would be set by cobra. Instead of adding the default values to each configuration that is being created in the tests, we should have a means to use a default configuration, e.g., by having a function `NewConfig`.

**Comment by anuraaga** [comment]
I suspect that it will also be natural to fix the issue that config can only be passed as global variables right now. This is awkward when running tests programmatically. I had added some config knobs related to HTTP previously but probably all the config should be merged into that, with the CLI passing in values same as it currently does.

**Comment by fzipi** [comment]
What about https://github.com/coreruleset/go-ftw/blob/e77aee39ed5be504ca91ea7e8755ecd86d070e7e/config/config.go#L88 for testing?

**Comment by anuraaga** [comment]
Global variables are awkward to use programmatically, these functions being called new but not returning any actual config seems to show that they aren't very idiomatic.

The biggest problem would be to run multiple ftw in parallel which seems reasonable to support.

**Comment by theseion** [comment]
What I was going for was a function that can is used by both the commands and the tests. With `NewConfigFromString` we would just repeat the default values for the tests. I don't like repeating defaults in different places, that always comes back to haunt you.

**Comment by fzipi** [comment]
Following #123 merge, I think we can now move to the second part of this fix. I guess adding the command line config variables that will affect all tests to the config is the right choice?

**Comment by theseion** [comment]
Yes, I think so.
